src/engraving/draw/qpainterprovider.cpp

```diff
@@ -170,22 +173,24 @@ void QPainterProvider::restore()
     m_brush = Brush::fromQBrush(m_painter->brush());
 }
 
-void QPainterProvider::setTransform(const QTransform& transform)
+void QPainterProvider::setTransform(const Transform& transform)
 {
     m_transform = transform;
-    m_painter->setTransform(transform);
+    m_painter->setTransform(Transform::toQTransform(m_transform));
 }
 
-const QTransform& QPainterProvider::transform() const
+const mu::Transform& QPainterProvider::transform() const
 {
-    return m_painter->transform();
+    return m_transform;
 }
 
 // drawing functions
 
-void QPainterProvider::drawPath(const QPainterPath& path)
+void QPainterProvider::drawPath(const PainterPath& path)
 {
-    m_painter->drawPath(path);
+    QPainterPath qpath;
+    DrawBufferJson::qPainterPathfromObj(DrawBufferJson::painterPathToObj(path), qpath);
```

Comment:
It is better to write a separate conversion function here, without using JSON (implementation with JSON is to different thing, it can be used as an example)